python/ray/serve/tests/test_deploy_2.py

```diff
@@ -239,6 +240,31 @@ def f():
         f.options(ray_actor_options={"runtime_env": {"working_dir": "."}}).deploy()
 
 
+def test_nonpicklable_user_config(serve_instance):
```

Comment:
hmm isn't this test more like `test_json_serialization_user_config` ? We haven't tested configs that's not JSON serializable in this PR

Reply:
That's true– I named it `nonpicklable` because of example `user_configs` from #25345 which failed because they were Pickled instead of serialized with JSON. For future readers though, I think `json_serialization` makes more sense.

Reply:
I updated the name.